mm2src/mm2_main/src/rpc/dispatcher/dispatcher.rs

```diff
@@ -244,7 +246,12 @@ async fn rpc_task_dispatcher(
             _ => MmError::err(DispatcherError::NoSuchMethod),
         },
         #[cfg(target_arch = "wasm32")]
-        _ => MmError::err(DispatcherError::NoSuchMethod),
+        wasm_only_methods => match wasm_only_methods {
+            "init_metamask::cancel" => handle_mmrpc(ctx, request, cancel_init_metamask).await,
+            "init_metamask::init" => handle_mmrpc(ctx, request, init_metamask).await,
```

Comment:
Do I understand correctly that GUI will first have to invoke `init_metamask::init` before attempting to initialize e.g. ETH in Metamask mode?

Reply:
Yes, right

Reply:
Ok, after thinking for some time, I think it's fine and even good. I hope it won't cause confusion or tricky state tracking on GUI side - will clarify it with Flutter team later 🙂